engine/src/main/java/org/hibernate/validator/scripting/AbstractCachingScriptEvaluatorFactory.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.scripting;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import javax.script.ScriptEngineManager;
+
+/**
+ * Basic cacheable factory responsible for the creation of {@link ScriptEvaluator}s. This
+ * class is thread-safe. Caches {@code ScriptEvaluator} when they are requested.
+ *
+ * @author Gunnar Morling
+ * @author Kevin Pollet &lt;kevin.pollet@serli.com&gt; (C) 2011 SERLI
+ * @author Marko Bekhta
+ */
+public abstract class AbstractCachingScriptEvaluatorFactory implements ScriptEvaluatorFactory {
+
+	/**
+	 * A cache of script evaluators (keyed by language name).
+	 */
+	private final ConcurrentMap<String, ScriptEvaluator> scriptEvaluatorCache = new ConcurrentHashMap<>();
+
+	/**
+	 * Retrieves a script executor for the given language.
+	 *
+	 * @param languageName The name of a scripting language as expected by {@link ScriptEngineManager#getEngineByName(String)}.
```

Comment:
Is the reference to `ScriptEngineManager` correct here, as we abstract from JSR 223 now?